Documentation/proposals/202403-prometheus-operator-cli.md

```diff
@@ -0,0 +1,92 @@
+
+## Why my Prometheus resource is not being created?
+
+People were struggling to troubleshoot why their Prometheus resources were not being created. They were not sure if the issue was with the Prometheus Operator or with the Prometheus resource itself.
+After long investigation, they found out that the Prometheus Operator were only watching the namespace where the Prometheus Operator was deployed, and the Prometheus resource was being created in a different namespace.
+
+## Why my ServiceMonitor is not being created?
+
+People were struggling to troubleshoot why their ServiceMonitor resources were not being created. They were not sure if the issue was with the Prometheus Operator or with the ServiceMonitor resource itself.
+
+After long investigation, they found out that the matchingLabels field in the ServiceMonitor resource was not matching the labels of the target Service.
+
+These are just some common issues that users reported, but there are many other issues that users face when managing Prometheus Operator resources, and we believe that a CLI tool can help users to manage Prometheus Operator resources more easily and efficiently.
+
+# Pitfalls of the current solution
+
+At the moment people are struggling to manage Prometheus Operator resources, they have to manually create, update, and delete Prometheus Operator resources using `kubectl` or other tools, and troubleshooting and debugging Prometheus Operator resources is challenging.
+
+# Goals
+
+The main goal of this proposal is to create a new CLI tool to manage and operate Prometheus Operator resources.
+
+# Non-Goals
+
+- Replace any existing config management tools such as Kustomize or Helm.
+
+# Audience
+
+- Users that are using Prometheus Operator to monitor their Kubernetes clusters.
+
+# How
+
+We propose to create a new CLI tool allowing users to create, update, and delete Prometheus Operator resources using a simple CLI interface as well as troubleshoot and debug Prometheus Operator resources.
+
+The main idea is to distribute this CLI as a [kubectl plugin](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/) and the user experience should be similar to the `kubectl` CLI, as you can see in the example below:
+
+```bash
+kubectl prometheus-operator create prometheus --name my-prometheus --namespace my-namespace --replicas 2 --shards 2
+```
+
+## Where the code will live?
+
+The prometheus-operator-cli code will be placed in a dedicated repository under the [Prometheus organization](https://github.com/prometheus-operator), this will allow us to have a separate lifecycle for the Prometheus Operator and the Prometheus Operator CLI and the following benefits:
+
+- We can have a dedicated team to maintain the Prometheus Operator CLI
+- We can use the Prometheus Operator client as a customer and be more careful with the changes that we make in the Prometheus Operator codebase
+- We can have a dedicated release cycle for the Prometheus Operator CLI
```

Comment:
1) Can also be solved with CODEOWNERS

2) I'm not sure if I'm misunderstanding something here, but if we want to be more careful with Prometheus-Operator codebase as a library, I think it makes more sense to keep the CLI in the same repo as prometheus-operator... since a breaking change would be caught by the IDE. Requiring bumps in go modules would make it a lot slower to catch breaking changes. 

3) Perfect 👍  but you already mentioned that 3 lines above